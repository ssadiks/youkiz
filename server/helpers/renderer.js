import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import serialize from 'serialize-javascript';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Routes from '../../src/Routes';
import HeaderContainer from '../../src/components/container/HeaderContainer/HeaderContainer';

const renderer = (req, store) => {
  const muiTheme = getMuiTheme({
    userAgent: req.headers['user-agent']
  });

  const content = renderToString(
    <Provider store={store}>
      <MuiThemeProvider muiTheme={muiTheme}>
        <StaticRouter location={req.path} context={{}}>
          <div>
            {renderRoutes(Routes)}
          </div>
        </StaticRouter>
      </MuiThemeProvider>
    </Provider>
  );

  return `
    <html>
      <head>
        <link href="css/style.css" rel="stylesheet">
      </head>
      <body>
        <div id="root">${content}</div>
        <script>
          window.INITIAL_STATE = ${serialize(store.getState())}
        </script>
        <script src="js/vendors.js"></script>
        <script src="js/bundle.js"></script>
      </body>
    </html>
  `;
};

export default renderer;
