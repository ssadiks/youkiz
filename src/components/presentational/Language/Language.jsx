import React from 'react';
import PropTypes from 'prop-types';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import { LOCALES } from '../../../constants';

const Language = props => (
  <SelectField
    value={props.locale}
    onChange={props.onChangeLocale}
    floatingLabelText="Country"
    className="country"
  >
    {
      LOCALES.map(locale => (
        <MenuItem
          key={locale}
          value={locale}
          primaryText={locale}
        />
      ))
    }

  </SelectField>
);

Language.PropTypes = {
  onChangeLocale: PropTypes.func.isRequired,
  locale: PropTypes.string.isRequired
};

export default Language;
