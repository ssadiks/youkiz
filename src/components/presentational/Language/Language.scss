.language {
  position: absolute;
  top: 50%;
  right: 1rem;
  z-index: 9999;
  transform: translateY(-50%);

  .flag-select .arrow-down {
    display: none;
  }
}
