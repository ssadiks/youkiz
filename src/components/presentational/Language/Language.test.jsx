import React from 'react';
import renderer from 'react-test-renderer';
import Language from './Language';

jest.mock('material-ui/SelectField', () => 'SelectField');
jest.mock('material-ui/MenuItem', () => 'MenuItem');

describe('Language render', () => {
  it('renders correctly', () => {
    const locale = 'es';
    const tree = renderer.create(
      <Language
        onChangeLocale={() => {}}
        locale={locale}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
