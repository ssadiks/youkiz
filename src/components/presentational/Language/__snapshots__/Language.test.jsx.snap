// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language render renders correctly 1`] = `
<SelectField
  className="country"
  floatingLabelText="Country"
  onChange={[Function]}
  value="es"
>
  <MenuItem
    primaryText="en"
    value="en"
  />
  <MenuItem
    primaryText="fr"
    value="fr"
  />
  <MenuItem
    primaryText="es"
    value="es"
  />
</SelectField>
`;
